plans/phase-1/multigrescluster-cr-definition.md

```diff
@@ -0,0 +1,713 @@
+
+While the primary workflow involves defining the entire topology within the `MultigresCluster` CR, the design also accommodates advanced use cases. Users will have the option to provision and manage child resources independently by setting a `managed: false` flag, allowing for greater flexibility and integration with existing infrastructure.
+
+
+### User Stories
+
+As a platform operator, I want to define my entire multi-cell, sharded Multigres topology in a single YAML file and apply it with `kubectl`. The system should then automatically provision all the necessary components, connect them, and bring the database cluster online without further manual intervention.
+
+### Operator Design and Architecture
+
+The operator's core responsibility is to translate the high-level state defined in the `MultigresCluster` CR into a set of child CRs. Each child CR represents a distinct component of the cluster, such as a cell or a database shard. The controllers for these child CRs are then responsible for creating and managing the low-level Kubernetes resources (like `StatefulSets`, `Deployments`, and `Services`) required to run the system.
+
+### API Definition (`MultigresCluster` CRD)
+
+The user interacts with the operator primarily through the `MultigresCluster` custom resource. This single manifest serves as the source of truth for the entire cluster. 
+
+> Please note that this sample does not contain the tablegroups or shard definitions, which are discussed later in this document.
+
+```yaml
+apiVersion: [multigres.com/v1alpha1](https://multigres.com/v1alpha1)
+kind: MultigresCluster
+metadata:
+  name: example-multigres-cluster
+  namespace: multigres
+spec:
+  # globalTopoServer defines the topology server for the entire cluster.
+  # Required.
+  globalTopoServer:
+    # If 'managed' is true, the operator will create and manage the global EtcdLockserver CR.
+    # Optional. Defaults to 'true'.
+    managed: true
+    # The root path in etcd where this cluster's topology data will be stored.
+    # Required.
+    rootPath: "/multigres/global"
+    # Defines the spec for the etcd cluster that will be managed by this operator.
+    # This is used as a template if 'managed' is true.
+    managedSpec:
+      image: quay.io/coreos/etcd:v3.5.17
+      replicas: 3
+      dataVolumeClaimTemplate:
+        accessModes: ["ReadWriteOnce"]
+        resources:
+          requests:
+            storage: "10Gi"
+    # Optional. Use this if 'managed' is false or to override the managed spec.
+    # external:
+    #   address: "my-external-etcd-client.etcd.svc:2379"
+
+  # admin provides a template for the MultiAdmin component.
```

Comment:
what's MultiAdmin? 🤔 I don't see that in [the architecture docs](https://multigres.com/docs/architecture)

Reply:
It is one of the [components](https://github.com/multigres/multigres/blob/main/Makefile#L65) of multigress. A web UI to administer the cluster.